lib/Conversion/TorchToLinalg/Pooling.cpp

```diff
@@ -228,6 +228,149 @@ class ConvertAtenAdaptiveAvgPool2dOp
 };
 } // namespace
 
+namespace {
+class ConvertAtenAvgPool2dOp : public OpConversionPattern<AtenAvgPool2dOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenAvgPool2dOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+    Location loc = op->getLoc();
+    Value self = adaptor.self();
+
+    Type elementType = self.getType().cast<RankedTensorType>().getElementType();
+    if (!elementType.isa<mlir::FloatType>())
+      return op.emitError("unimplemented: non-floating point type");
+
+    // Pattern match against the op's original operands, because otherwise we
+    // will get the lowered version of the operands which is harder to pattern
+    // match.
+    SmallVector<int64_t, 2> kernelSizeInts;
+    if (!matchPattern(op.kernel_size(), m_TorchConstantIntList(kernelSizeInts)))
+      return rewriter.notifyMatchFailure(
+          op, "only support constant int kernel size");
+    SmallVector<int64_t, 2> strideInts;
+    if (!matchPattern(op.stride(), m_TorchConstantIntList(strideInts)))
+      return rewriter.notifyMatchFailure(op,
+                                         "only support constant int strides");
+    SmallVector<int64_t, 2> paddingInts;
+    if (!matchPattern(op.padding(), m_TorchConstantIntList(paddingInts)))
+      return rewriter.notifyMatchFailure(op,
+                                         "only support constant int paddings");
+
+    // The ceil_mode should be a constant `False`.
```

Comment:
Why the `ceil_mode` should be False? Is it a TODO? https://pytorch.org/docs/stable/generated/torch.nn.AvgPool2d.html I see that ceil_mode = True is also supported.